### PR TITLE
MM-2998 Add telemetry for product notices

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
   * Add  track event when user clicks on action button of notice

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-29998

#### Related Pull Requests
https://github.com/mattermost/mattermost-redux/pull/1277

### CODE REVIEW

**Review by nevyangelova — APPROVED** [review]
LGTM!

File: components/product_notices_modal/product_notices.test.tsx [hunk 1]
```diff
@@ -6,10 +6,18 @@ import {shallow} from 'enzyme';
 
 import GenericModal from 'components/generic_modal';
 import {isDesktopApp, getDesktopVersion} from 'utils/user_agent';
+import {trackEvent} from 'actions/diagnostics_actions.jsx';
```
Comment (nevyangelova) [inline components/product_notices_modal/product_notices.test.tsx:9]: This seems to be removed?

Comment (sudheerDev) [reply components/product_notices_modal/product_notices.test.tsx]: Aah, it is rename to telemetry actions

Comment (sudheerDev) [reply components/product_notices_modal/product_notices.test.tsx]: Thank you

File: components/product_notices_modal/product_notices_modal.tsx [hunk 2]
```diff
@@ -186,6 +188,11 @@ export default class ProductNoticesModal extends React.PureComponent<Props, Stat
         return null;
     }
 
+    private trackClickEvent = () => {
+        const presentNoticeInfo = this.state.noticesData[this.state.presentNoticeIndex];
+        trackEvent('ui', `notice_click_${presentNoticeInfo.id}`);
```
Comment (Willyfrog) [inline components/product_notices_modal/product_notices_modal.tsx:193]: not to change it right now, but feels like `'ui'` should really be a constant so we use the same string throughout the code prevent typos

**Review by Willyfrog — APPROVED** [review]
left a comment, but I don't think it is in the scope of the change, something to consider for the future. LGTM

File: components/product_notices_modal/product_notices_modal.tsx (hunk 2 above)
Comment (sudheerDev) [reply components/product_notices_modal/product_notices_modal.tsx:193]: Ya, i agree. All events strings need to be changed to constants. I think we can run a campaign on this

**Review by jgilliam17 — APPROVED** [review]
Thank you @sudheerDev 
Per our discussion QA will test telemetry for product notices after merge.

**Comment by amyblais** [comment]
@esethna @sudheerDev I added the "Docs/Needed" label as I wasn't sure if this needs docs for https://docs.mattermost.com/administration/telemetry.html. Please take a look.

**Comment by esethna** [comment]
@sudheerDev can you help submit a PR as needed?

**Comment by sudheerDev** [comment]
> @sudheerDev can you help submit a PR as needed?

@amyblais @esethna Will do

**Pull request merged**